### PR TITLE
Missing snowflake view lineage [sc-11489]

### PR DESCRIPTION
**Pull request by alyiwang**
### 🤔 Why?

the `access_history` doesn’t contain any table to view lineage, since creating a view by query does not actually write any data.

### 🤓 What?

- use the `object_dependencies` view to get table to view direct relationship. https://docs.snowflake.com/en/user-guide/object-dependencies.html#label-object-dependencies-view-query

### 🧪 Tested?

tested against our snowflake instance and verified the output JSON

### CODE REVIEW

**Review by mars-lan — COMMENTED** [review]
For consistency, can we also add a `enable_view_lineage` option like BQ? https://github.com/MetaphorData/connectors/tree/main/metaphor/bigquery/lineage

**Comment by alyiwang** [comment]
> For consistency, can we also add a `enable_view_lineage` option like BQ? https://github.com/MetaphorData/connectors/tree/main/metaphor/bigquery/lineage

Updated

File: metaphor/snowflake/lineage/README.md [hunk 1]
```diff
@@ -11,6 +11,12 @@ Create a dedicated user & role based on the [Setup](../README.md#Setup) guide fo
 The config file inherits all the required and optional fields from the general Snowflake connector [Config File](../README.md#config-file). In addition, you can specify the following configurations:
 
 ```yaml
+# (Optional)Whether to enable finding view lineage from object dependencies, default True
```
Comment (mars-lan) [inline metaphor/snowflake/lineage/README.md:14]:
```suggestion
# (Optional) Whether to enable finding view lineage from object dependencies, default True
```

File: metaphor/snowflake/lineage/README.md [hunk 2]
```diff
@@ -11,6 +11,12 @@ Create a dedicated user & role based on the [Setup](../README.md#Setup) guide fo
...
+enable_view_lineage: bool = True
+
+# (Optional) Whether to enable finding table lineage information from access history and query history, default True
```
Comment (mars-lan) [inline metaphor/snowflake/lineage/README.md:17]:
```suggestion
# (Optional) Whether to enable finding table lineage information from access history and query history, default True.
```

**Review by mars-lan — COMMENTED** [review]
Otherwise LGTM

**Review by mars-lan — APPROVED** [review]

**Pull request merged**